stdlib/public/core/Sequence.swift

```diff
@@ -583,6 +583,18 @@ extension SequenceType where Generator.Element : Equatable {
     return split(maxSplit, allowEmptySlices: allowEmptySlices,
       isSeparator: { $0 == separator })
   }
+
+  /// Return an `Array` containing the elements of `self`
+  @warn_unused_result
+  public func uniq() -> [Generator.Element] {
+    return reduce(Array<Generator.Element>()) { (arr, element) -> [Generator.Element] in
```

Comment:
The dictionary is probably unnecessary - there's no need to store the `Bool` values. You could do the same with just a `Set`:

```swift
extension SequenceType where Generator.Element : Hashable {
  
  public func uniq() -> [Generator.Element] {
    var prevs: Set<Generator.Element> = []
    return filter { el in
      if prevs.contains(el) { return false }
      prevs.insert(el)
      return true
    }
  }
}
```

Which is maybe a little more efficient. Both the `Dictionary` and `Set` versions translate easily into a lazy method:

```swift
extension LazySequenceType where Generator.Element : Hashable {
  
  public func uniq() -> LazyFilterSequence<Self> {
    var prevs: Set<Generator.Element> = []
    return lazy.filter { el in
      if prevs.contains(el) { return false }
      prevs.insert(el)
      return true
    }
  }
}
```

However, there's another problem with these implementations: they rely on side-effects from `filter`'s closure, which is generally discouraged (I think). For instance, the function would be broken if the closure was evaluated more than once, or in a different order. So it's maybe best to have a more imperative `for` loop:

```swift
extension SequenceType where Generator.Element : Hashable {
  
  public func uniq() -> [Generator.Element] {
    var prevs: Set<Generator.Element> = []
    var uniqs: [Generator.Element] = []
    for el in self {
      if !prevs.contains(el) {
        prevs.insert(el)
        uniqs.append(el)
      }
    }
    return uniqs
  }
}
```

But then the lazy version would need to be more complex as well. (You'd probably have to write a custom struct, similar to the standard library's `LazyFilterSequence`. [This is the kind of thing I mean](https://github.com/oisdk/SwiftSequence/blob/master/SwiftSequence/Categorise.swift#L101))

Considering all that, I'm not sure what the function or method would add over just using the standard `Set` initialiser. The order of the elements is retained, and there's a possible lazy version, but it's a lot of extra complexity for maybe not much extra functionality.